cmd/prometheus/main.go

```diff
@@ -294,8 +296,8 @@ func main() {
 	}
 
 	// Unregister the default GoCollector, and reregister with our defaults.
-	if prometheus.Unregister(collectors.NewGoCollector()) {
-		prometheus.MustRegister(
+	if registry.Unregister(collectors.NewGoCollector()) {
+		registry.MustRegister(
```

Comment:
You also need to register a `NewProcessCollector`. That's the other collector that is registered with the default registry, see https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#pkg-variables